core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java

```diff
@@ -301,6 +321,27 @@ public void send(
       String format,
       Predicate<MatchPlayer> filter,
       @Nullable SettingValue type) {
+
+    // Call a chatevent for each message, will cancel the message if the event is canceled by
+    // anything except PGM
+    if (sender != null) {
+      AsyncPlayerChatEvent event =
+          new AsyncPlayerChatEvent(
+              false,
+              sender.getBukkit(),
+              message,
+              match.getPlayers().stream()
+                  .filter(filter)
+                  .map(MatchPlayer::getBukkit)
+                  .collect(Collectors.toSet()));
+
+      messageCache.put(event, "boring message");
+
+      match.callEvent(event);
+
+      if (event.isCancelled()) return;
```

Comment:
We need to respect any changes to the event, say if the message is changed or the viewers are changed. So more than just check if cancelled.

Reply:
Of course. I totally forgot that haha. Thanks!